### ISSUE TITLE
setRowStyle function not working

### ISSUE THREAD

**Issue by eduardo-mior** [qax-os/excelize#1129]
**Description**
setRowStyle function not working.

I need to define the style of a line but this is not working in Excel and GoogleSheets, interestingly in LibreOffice Calcs it works.

**Code example:**
```golang
xlsx.SetSheetName("Sheet1", "Report")

titleStyle, err := xlsx.NewStyle(&excelize.Style{
	Font: &excelize.Font{
		Bold: true,
		Size: 20,
	},
	Alignment: &excelize.Alignment{
		Vertical: "center",
	},
})
if err != nil {
	return "", nil, err
}

if err := xlsx.SetCellValue("Report", "A1", "Customers Report"); err != nil {
	return "", nil, err
}

if err := xlsx.SetRowStyle("Report", 1, 1, titleStyle); err != nil {
	return "", nil, err
}

if err := xlsx.SetRowHeight("Report", 1, 30); err != nil {
	return "", nil, err
}

if err := xlsx.MergeCell("Report", "A1", "Z1"); err != nil {
	return "", nil, err
}

xlsx.SaveAs("teste.xlsx")
```

**Describe the results you received:**
GoogleSheets and Excel:
![image](https://user-images.githubusercontent.com/32434418/150829442-eb0708a6-98f3-4bc8-89d7-8a0c6d216bd7.png)

**Describe the results you expected:**
LibreOffice:
![image](https://user-images.githubusercontent.com/32434418/150829418-8d563c71-6749-4445-957b-6ec888ef895c.png)

**Output of `go version`:**
```text
go version go1.17.3
```

**Excelize version or commit ID:** 2.5.0

**Environment details (OS, Microsoft Excel™ version, physical, etc.):** 
MacOS Catalina AND Linux Zorion OS
GoogleSheets (online), Libre Office (linux), Excel 2021 (macos from app store)

**Comment by xuri** [comment]
Thanks for your issue. Please set the style for the cells in the merged range like this:

```go
err := xlsx.SetCellStyle("Report", "A1", "Z1", titleStyle)
```

**Comment by eduardo-mior** [comment]
Many thanks for the reply.

**Comment by eduardo-mior** [comment]
Do you plan to resolve this issue in a future release? I noticed that you closed the issue but didn't leave any other comments.

**Comment by xuri** [comment]
I have fixed it, please upgrade to the master branch code, and this patch will be released in the next version.

**Comment by kukaki** [comment]
@xuri

SetRowStyle is not replacing the style of a cell (part of the row) which was previously set using SetCellStyle.

For eg. **Let's say I have set style to column A using SetColStyle. Now if I set style to row 1 using SetRowStyle, the row style is applied to entire row 1 except cell A1 (col style prevailing here).**

i.e. SetRowStyle is setting style to only the cells which do not have a previous style.

It is not intended, right?

**Comment by xuri** [comment]
Hi @kukaki, sorry for my late reply, I have made the `SetRowStyle` overwrite style of the cells on commit [`773d4af`](https://github.com/xuri/excelize/tree/773d4afa32a55349a7b178c4c76d182f9ed0221f), please upgrade to the master branch code, and this patch will be released in the next version.

**Comment by xuri** [comment]
Hi @eduardo-mior, I also notice that according to the [benchmark report](https://github.com/xuri/excelize/wiki#performance-figures) of v2.6.0, this fix (commit: [`156bf6d`](https://github.com/xuri/excelize/tree/156bf6d16ecbd5257d81e781138eaaaf357ffbec)) slows down the merging cell about 400 times.

**Comment by xuri** [comment]
Hi @eduardo-mior, I also notice that according to the [benchmark report](https://github.com/xuri/excelize/wiki#performance-figures) of v2.6.0, this fix (commit: [`156bf6d`](https://github.com/xuri/excelize/tree/156bf6d16ecbd5257d81e781138eaaaf357ffbec)) slows down set sheet row and the merging cell about 400 times.

**Comment by xuri** [comment]
Hi @eduardo-mior, I have improved performance for the set sheet row and the merging cells on commit [`eed431e`](https://github.com/xuri/excelize/tree/eed431e0fc2f61b13e7745857a41cb47d9f7f810), fixed performance impact when resolving this issue, and this patch will be released in the next version.

**Comment by michaelperel** [comment]
@xuri I am not sure if this is the same issue (I recently started using the library), however:

In the latest v2 release, when I set a row style for 1 row, and then write data into row 2, 3, 4 and so on, the style from row 1 is applied to all rows. I would expect the style to only apply to row 1, not subsequent rows.

In master (commit: eed431e0fc2f61b13e7745857a41cb47d9f7f810), the behavior is as I would have expected - If I set the style for row 1 and then write row 2, the style from row 1 does not apply to row 2.

So that is to say, master is working great for me!
